src/macaron/slsa_analyzer/package_registry/deps_dev.py

```diff
@@ -50,34 +47,77 @@ def get_package_info(purl: str) -> dict | None:
                 f'The "url_netloc" key is missing in section [{section_name}] of the .ini configuration file.'
             )
         url_scheme = section.get("url_scheme", "https")
-        purl_endpoint = section.get("purl_endpoint")
-        if not purl_endpoint:
+
+        api_endpoint = section.get("api_endpoint")
+        if not api_endpoint:
             raise APIAccessError(
-                f'The "purl_endpoint" key is missing in section [{section_name}] of the .ini configuration file.'
+                f'The "api_endpoint" key is missing in section [{section_name}] of the .ini configuration file.'
             )
-
-        path_params = "/".join([purl_endpoint, encode(purl, safe="")])
-        try:
-            url = urllib.parse.urlunsplit(
-                urllib.parse.SplitResult(
+        endpoint_path = [api_endpoint]
+        if path:
+            endpoint_path.append(path)
+        if not purl:
+            try:
+                return urllib.parse.SplitResult(
                     scheme=url_scheme,
                     netloc=url_netloc,
-                    path=path_params,
+                    path="/".join(endpoint_path),
                     query="",
                     fragment="",
                 )
+            except ValueError as error:
+                raise APIAccessError("Failed to construct the API URL.") from error
+
+        purl_endpoint = section.get("purl_endpoint")
+        if not purl_endpoint:
+            raise APIAccessError(
+                f'The "purl_endpoint" key is missing in section [{section_name}] of the .ini configuration file.'
+            )
+        endpoint_path.insert(1, purl_endpoint)
+        try:
+            return urllib.parse.SplitResult(
+                scheme=url_scheme,
+                netloc=url_netloc,
+                path="/".join(endpoint_path),
+                query="",
+                fragment="",
             )
         except ValueError as error:
             raise APIAccessError("Failed to construct the API URL.") from error
 
+    @staticmethod
+    def get_package_info(purl: str) -> dict:
+        """Check if the package identified by the PackageURL (PURL) exists and return its information.
+
+        Parameters
+        ----------
+        purl: str
+            The PackageURL (PURL).
+
+        Returns
+        -------
+        dict
+            The package metadata.
+
+        Raises
+        ------
+        APIAccessError
+            If the service is misconfigured, the API is invalid, a network error happens,
+            or unexpected response is returned by the API.
+        """
+        if "%" in purl:
```

Comment:
It would be good to have a comment or an example here to explain why we need to convert all "%xx" to their character equivalent. 
If I understand it correctly, if we don't convert "%xx", the call to quote / encode below will convert % to "%25", which is not what we want.